install-build-deps.sh

```diff
@@ -4,35 +4,45 @@ source /etc/os-release
 
 # The first line for each distro installs a build dependency.
 # The second line installs extra packages for `make test`.
+test_id_version() {
+  case "$1$2" in
+  ubuntu20.*)
+    apt-get install -y git cmake libssl-dev zlib1g-dev gcc g++ g++-10
+    apt-get install -y file bsdmainutils
+    ;;
+  ubuntu22.* | debian11)
+    apt-get install -y git cmake libssl-dev zlib1g-dev gcc g++
+    apt-get install -y file bsdmainutils
+    ;;
+  fedora*)
+    dnf install -y git gcc-g++ cmake openssl-devel zlib-devel
+    dnf install -y glibc-static file libstdc++-static diffutils
+    ;;
+  opensuse-leap*)
+    zypper install -y git make cmake zlib-devel libopenssl-devel gcc-c++ gcc11-c++
+    zypper install -y glibc-devel-static tar diffutils util-linux
+    ;;
+  opensuse-tumbleweed*)
+    zypper install -y git make cmake zlib-devel libopenssl-devel gcc-c++
+    zypper install -y glibc-devel-static tar diffutils util-linux
+    ;;
+  gentoo*)
+    emerge dev-vcs/git dev-util/cmake sys-libs/zlib
+    ;;
+  arch*)
+    pacman -S --needed --noconfirm base-devel zlib openssl cmake util-linux git
+    ;;
+  *)
+    echo "Error: don't know anything about build dependencies on $1 $2"
+    return 1
+    ;;
+  esac
+}
 
-case "$ID$VERSION_ID" in
-ubuntu20.*)
```

Comment:
Then I think we should simply add `pop20.*` here and `pop22.*` below.

Reply:
Doesn't that leave behind possible other derivatives?

From my point of view, I see these changes as way to stop the script from increasing in scope with every derivative someone submits a PR for.

I think that anyone tech-savvy enough to compile mold from source, and happens to use a derivative OS, would be able to copy the relevant dependency command out from `install-build-deps.sh` for their system.

Reply:
I don't think there are too many derived distros. Do you have anything other than Pop!_OS? Also, users of minor distros are likely to be tech savvy, so they can easily run commands by hand. To make it crystal clear that what commands have to be run, I want to keep this script as simple as possible, so I don't want to use anything other than table lookup.

Reply:
That makes sense to me, thank you for the clarification. Copying the apt command is what I did when installing v1.4.0.

In light of the expectations of small distro users and the goal of keeping this script simple, I think it would be best not to merge this PR at all

Reply:
I'm fine to add `pop20.*` and `pop22.*` to the table. As long as we don't add any logic, adding new entries to these pattern matches doesn't increase complexity.

Reply:
If that's the case, I'll make a separate PR for that; modifying this PR and its commit history to reflect only that small change seems misleading and disconnected from the earlier discussion.

Reply:
OK, then let me close this PR.

Reply:
I added pop-20 and pop-22 to the list. I noticed that `ID_LIKE` wasn't suitable for our purpose because it doesn't specify a version number. For example, `ubuntu` is defined as a `debian`-like in their `/etc/os-release`, but it doesn't say anything about what version it corresponds to. Trying to install as if it's `debian-20` isn't right (the latest debian version is 11.) So just enumerating all possible OSes and version numbers seems to be the best we can do.

Reply:
That was definitely a problem I failed to consider. Thank you for finding a solution for this that works for your project, I enjoyed being a (small) part of the process :)